Next version (primeng)

jma opened the pull request:
- Closes #3758.

Garfield-fr committed:
chore(schema): use select instead of selectWithSort

Co-Authored-by: Bertrand Zuchuat <bertrand.zuchuat@rero.ch>

Garfield-fr committed:
chore(schema): update schemas

Co-Authored-by: Bertrand Zuchuat <bertrand.zuchuat@rero.ch>

jma committed:
ui: uses primeng for the frontend

* Adds missing format alias on the the REST API.
* Adds placeholder on the contries editor.
* All document detailed view tabs is done in angular.
* Removes useless document views.
* Removes useless document jinja template filters and their corresponding tests.
* Adds layer to css styles to avoid primeng and bootstrap conflicts.

Co-Authored-by: Johnny Mariéthoz <Johnny.Mariethoz@rero.ch>

jma committed:
primeng: fix patron editor

* Removes css class for the professional view.
* Adapts the main search component to the new ui version.
* Fixes several details in the patron editor.

Co-Authored-by: Johnny Mariéthoz <Johnny.Mariethoz@rero.ch>
Co-Authored-by: Bertrand Zuchuat <bertrand.zuchuat@rero.ch>

ManaDeweerdt commented:
- [ ] Editor: cannot select a contribution role.

PascalRepond commented:
## PO tests (entites)

- [ ] Delete buttons should be hidden for remote entites (brief views)
- [ ] Buttons should all be same size (detailed view)
![image](https://github.com/user-attachments/assets/6f69bb97-c8e4-499c-b9cb-5c79dba598d6)

PascalRepond commented:
## PO tests (reports and stats)

- [ ] Fees brief view: some facets titles not translated (Amount recieved, Transaction_date)
- [ ] Stat reports editor: distribution selection doesn't work
- [ ] Stat reports editor: sometimes "Period" field should be hidden from editor for certain indicators (e.g.: Catalog->Number of documents)

PascalRepond commented:
## PO tests (admin menu)

- Editors: when all fields have been added, there is still a box for essential fields, this could be hidden.
![image](https://github.com/user-attachments/assets/fe6391b3-68a6-44e1-94d9-51cf90880a2e)
- Library editor: Button "Add" location is displayed even when menu is collapsed
![image](https://github.com/user-attachments/assets/24bfcdac-2a3d-4a20-b9e7-fe991991c806)

jma committed:
chore: acquisitions

* Fixes poetry shell plugins.
* Removes is_canceled, send_now, exange_rate, receipt_date useless acquistions fields.
* Moves order_date from order lines to orders.
* Removes order type as it is useless.
* Uses primeng css class in the JSONSchemas.
* Adds the number of children in the acquisition acounts serializer.
* Adjusts acquistion permissions.
* Removes toggle-switch type in JSONSchemas.
* Fixes the multi-select type in JSONSchemas.
* Fixes invalid user name regular expression.
* Uniformizes placeholders.

Co-Authored-by: Johnny Mariéthoz <Johnny.Mariethoz@rero.ch>
Co-Authored-by: Pascal Repond <pascal.repond@rero.ch>

PascalRepond commented:
## PO tests (acquisitions)

- [ ] When an order is fully recieved, the receipt line cannot be edited but the button is still available.

PascalRepond commented:
## PO tests (circulation)

- [ ] Add a fee modal: selecting type (dropdown) doesn't work on first click.

PascalRepond commented:
## PO tests - editors

### ILL request

- OK

### User/patron

### Exhibition/course

- OK

### Document

### Holdings

### Item

### Local entity

- OK

### Vendor

### Order

### Stat report config

### Circulation policy

### Item type

### Patron type

### Library

### Template

- [ ] If I want to load a template (e.g. from the document editor), the modal is small and the dropdown does not expand out of it.
![image](https://github.com/user-attachments/assets/942ec1b3-a1d4-4b3e-9867-88377ef4402f)

PascalRepond commented:
## PO tests (13.02)

- [ ] Public view: search bar too small
- [ ]

PascalRepond commented:
## Tests (21.02)

- [ ] User editor, when going back to patron editor, the user name is not displayed and the button says "create" instead of "edit"
- [ ] NOT-BLOCKING: Tooltips in editors  would be better if aligned to the left instead of centered
![image](https://github.com/user-attachments/assets/244cbf2c-16d8-4af0-bc85-740604b1aa64)
- [ ] NOT BLOCKING Document detail: The first item under "Other organisations" should not be expanded on page load
- [ ] Switch location permissions (regression): I used to be able to switch to a location for another library when I am system librarian or have rights to multiple libs. Now I can only switch to the currently active library.

PascalRepond committed:
chore: enhance editors UX

* Adapts certain classes in the jsonschemas for tailwind-css and for
better editors usability.
* Fixes some errors in the jsonschemas.

Co-Authored-by: Pascal Repond <pascal.repond@rero.ch>

jma committed:
chore: upgrade to angular 19

Co-Authored-by: Bertrand Zuchuat <bertrand.zuchuat@rero.ch>
Co-Authored-by: Johnny Mariéthoz <johnny.mariethoz@rero.ch>

PascalRepond committed:
chore(editors): fix css classes for ui prefixes

- Also removes an incoherent placeholder (document_relation) and useless
wrappers.

Co-Authored-by: Pascal Repond <pascal.repond@rero.ch>

jma committed:
fix(ui): add ui css prefix

* Fixes layer order for admin view.
* Adds ui: css prefix to the angular components.

Co-Authored-by: Bertrand Zuchuat <bertrand.zuchuat@rero.ch>
Co-Authored-by: Johnny Mariéthoz <johnny.mariethoz@rero.ch>

PascalRepond commented:
## Tests Angular 19 (APRIL)

- [ ] REP: Props addonRight seems not translated (e.g. circ policy editor "day(s)")
- [ ] REP: Long editors: Display problems on screen width below lg
![image](https://github.com/user-attachments/assets/6f85c92c-0f4a-49b2-af36-a2d204146ecd)
- [ ] REP: Document editor, contrib/genreForm/subjects: props.filters.options seem not translated 
![image](https://github.com/user-attachments/assets/30c0d206-68fd-4a49-8d78-5dca1df598c6)
- [ ] REP: The menu to add a field is too small, it could be at least 2x as high. 
- [ ] REP: The menu to add a field moves in the page when scrolling ! It should be fixed
![image](https://github.com/user-attachments/assets/abe98f86-a83d-4319-900d-f92e77f0505c)
- [ ] MD: not redirected when we click on the field 'published in' (need to refresh). Same with 'supplement to', 'related to' etc. Instead, the record is displayed below. Ex : https://ilsdev.test.rero.ch/professional/records/documents/detail/2000094
![image](https://github.com/user-attachments/assets/347a92e1-a9bd-43f9-b504-1263b13d5b56)

jma committed:
fix(ui): revert bootstrap css replacement

Co-Authored-by: Johnny Mariéthoz <johnny.mariethoz@rero.ch>

Garfield-fr committed:
chore: update jsonschema

Co-Authored-by: Bertrand Zuchuat <bertrand.zuchuat@rero.ch>

PascalRepond committed:
fix(translations): extract editor fields "addon"

Co-Authored-by: Pascal Repond <pascal.repond@rero.ch>

PascalRepond committed:
chore(acq_accounts): adjust some fields' width

Co-Authored-by: Pascal Repond <pascal.repond@rero.ch>

PascalRepond commented:
## Test (deployment 14.04)

- [ ] REP: Patron link in patron card when doing a *direct checkin* doesn't work.
- [ ] WEP [not-blocking]: overflow of menu in message component in circulation view

## Check after translations extractions (REP)

- [ ] REP: Some props `addonRight` seems not translated (See circ policy editor "day(s)")
- [ ] REP: Missing translation in Babelio tab (https://ilsdev.test.rero.ch/aoste/documents/243 - "No additional information yet.")
- [ ] EWN: Item editor some elements are not translated: "Temporary item type"

ManaDeweerdt commented:
## Test (deployment 30.04)

- [ ] Patron account (public view): keep history toggle not taken into account. Historical always displayed.
